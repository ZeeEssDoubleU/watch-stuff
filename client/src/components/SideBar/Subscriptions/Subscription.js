import React from "react";
import { Menu, Image } from "semantic-ui-react";
// import styles
import "./Subscription.scss";

// TODO - Fix styling
const Subscription = props => {
	const channelUrl = `https://www.youtube.com/channel/${props.id}`;
	return (
		<a className="sidebar-item" href={channelUrl} target="_blank" rel="noopener noreferrer">
			<Menu.Item>
				<div className="sidebar-item-alignment-container">
					<Image src={props.icon} avatar />
					<span>{props.label}</span>
				</div>
			</Menu.Item>
		</a>
	);
};

export default Subscription;
