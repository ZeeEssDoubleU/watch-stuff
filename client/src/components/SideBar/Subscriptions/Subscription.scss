@import "../../../styles/shared.scss";

.sidebar-item {
	.sidebar-item-alignment-container {
		display: flex;
		align-items: center;

		.avatar {
			margin-right: 20px;
			width: 25px;
			height: 25px;
		}
	}
}
