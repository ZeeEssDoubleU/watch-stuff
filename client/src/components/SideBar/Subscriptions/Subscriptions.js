import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import { Divider } from "semantic-ui-react";
// import components
import Subscription from "./Subscription";
import SideBarHeader from "../SideBarHeader";
// import actions / reducers / sagas
import { selector_subscriptions } from "../../../store/reducers/user";

// TODO - Update Subscriptions code to actually pull activity (broadcasting, new vids, etc) and update SideBar with data
const Subscriptions = props => {
	const { subscriptions } = props;

	if (Object.keys(subscriptions).length === 0) return <div />;

	const displaySubs = Object.keys(subscriptions).map(channelId => {
		const { channelTitle, channelIcon } = subscriptions[channelId];
		return (
			<Subscription
				id={channelId}
				label={channelTitle}
				icon={channelIcon}
				key={channelId}
			/>
		);
	});

	return (
		<>
			<SideBarHeader title="Subscriptions" />
			{displaySubs}
			<Divider />
		</>
	);
};

const mapStateToProps = state => ({
	subscriptions: selector_subscriptions(state),
});

export default connect(
	mapStateToProps,
	null,
)(Subscriptions);
