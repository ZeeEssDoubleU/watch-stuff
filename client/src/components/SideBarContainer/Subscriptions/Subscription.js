import React from "react";
import { Menu, Icon, Image } from "semantic-ui-react";

import './Subscription.scss';

const Subscription = props => {
	let rightElement = null;
	if (props.broadcasting) {
		rightElement = <Icon name="signal" />;
	} else if (props.amountNewVideos) {
		rightElement = <span className="new-videos-count">{props.amountNewVideos}</span>;
	}

	return (
		<Menu.Item>
			<div className="subscription">
				<div>
					<Image src="http://via.placeholder.com/28x28" avatar />
					<span>{props.label}</span>
				</div>
				{rightElement}
			</div>
		</Menu.Item>
	);
};

export default Subscription;
