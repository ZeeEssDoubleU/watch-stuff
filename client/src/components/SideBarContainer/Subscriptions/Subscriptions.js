import React, { Component } from "react";
import { Divider } from "semantic-ui-react";

import Subscription from "./Subscription";
import SideBarHeader from "../SideBarHeader";

class Subscriptions extends Component {
	render() {
		return (
			<>
				<SideBarHeader title="Subscriptions" />
				<Subscription label="Music Channel" broadcasting />
				<Subscription label="Coursea" amountNewVideos={10} />
				<Subscription label="TEDx Talks" amountNewVideos={23} />
				<Subscription label="Standford iOs" amountNewVideos={4} />
				<Subscription label="Udacity" amountNewVideos={114} />
				<Divider />
			</>
		);
	}
}

export default Subscriptions;
