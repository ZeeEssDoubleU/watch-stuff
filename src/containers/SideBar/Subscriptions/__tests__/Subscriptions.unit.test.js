import React from "react";
import { shallow } from "enzyme";

import Subscriptions from "../Subscriptions.js";

test("renders Subscriptions", () => {
	const wrapper = shallow(<Subscriptions />);
	expect(wrapper).toMatchSnapshot();
});
