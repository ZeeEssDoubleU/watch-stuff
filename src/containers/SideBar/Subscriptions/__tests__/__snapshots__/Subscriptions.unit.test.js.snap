// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Subscriptions 1`] = `
<Fragment>
  <SideBarHeader
    title="Subscriptions"
  />
  <Subscription
    broadcasting={true}
    label="Music Channel"
  />
  <Subscription
    amountNewVideos={10}
    label="Coursea"
  />
  <Subscription
    amountNewVideos={23}
    label="TEDx Talks"
  />
  <Subscription
    amountNewVideos={4}
    label="Standford iOs"
  />
  <Subscription
    amountNewVideos={114}
    label="Udacity"
  />
  <Divider />
</Fragment>
`;
